restli-tools/src/test/java/com/linkedin/restli/tools/snapshot/check/TestPegasusSchemaSnapshotCompatibilityChecker.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2020 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.restli.tools.snapshot.check;
+
+import com.linkedin.data.schema.compatibility.CompatibilityOptions;
+import com.linkedin.restli.tools.compatibility.CompatibilityInfoMap;
+import com.linkedin.restli.tools.idlcheck.CompatibilityInfo;
+import com.linkedin.restli.tools.idlcheck.CompatibilityLevel;
+import java.io.File;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+public class TestPegasusSchemaSnapshotCompatibilityChecker
+{
+  private final String FS = File.separator;
+  private String testDir = System.getProperty("testDir", new File("src/test").getAbsolutePath());
+  private String snapshotDir = testDir + FS + "pegasusSchemaSnapshot";
+
+  @Test
+  public void testCompatiblePegasusSchemaSnapshot()
+  {
+    PegasusSchemaSnapshotCompatibilityChecker checker = new PegasusSchemaSnapshotCompatibilityChecker();
+    checker.checkPegasusSchemaCompatibility(snapshotDir + FS + "BirthInfo.pdl", snapshotDir + FS + "compatibleSchemaSnapshot/BirthInfo.pdl",
+        CompatibilityLevel.EQUIVALENT, CompatibilityOptions.Mode.DATA);
+    CompatibilityInfoMap infoMap = checker.getInfoMap();
+    Assert.assertTrue(infoMap.isModelCompatible(CompatibilityLevel.EQUIVALENT));
+  }
+
+  @Test
+  public void testIncompatiblePegasusSchemaSnapshot()
```

Comment:
I think it might be better to abstract test input out from the test logic.
One example is the "BirthInfo"

Reply:
Used DataProvider instead